textual/src/toga_textual/window.py

```diff
@@ -1,17 +1,96 @@
+from rich.text import Text
+
+from textual.app import RenderResult
+from textual.reactive import Reactive
 from textual.screen import Screen as TextualScreen
-from textual.widgets import Header as TextualHeader
+from textual.widget import Widget as TextualWidget
 
 from .container import Container
 
 
+class CloseIcon(TextualWidget):
+    DEFAULT_CSS = """
+    CloseIcon {
+        dock: left;
+        padding: 0 1;
+        width: 4;
+        content-align: left middle;
+    }
+    """
+
+    def render(self) -> RenderResult:
+        return "⭘"
+
+
+class TitleSpacer(TextualWidget):
+    DEFAULT_CSS = """
+    TitleSpacer {
+        dock: right;
+        padding: 0 1;
+        width: 4;
+        content-align: right middle;
+    }
+    """
+
+    def render(self) -> RenderResult:
+        return ""
+
+
+class TitleText(TextualWidget):
+    DEFAULT_CSS = """
+    TitleText {
+        content-align: center middle;
+        width: 100%;
+    }
+    """
+    text: Reactive[str] = Reactive("")
+
+    def __init__(self, text):
+        super().__init__()
+        self.text = text
+
+    def render(self) -> RenderResult:
+        return Text(self.text, no_wrap=True, overflow="ellipsis")
+
+
+class TitleBar(TextualWidget):
+    DEFAULT_CSS = """
+    TitleBar {
+        dock: top;
+        width: 100%;
+        background: $foreground 5%;
+        color: $text;
+        height: 1;
+    }
+    """
+
+    def __init__(self, title):
+        super().__init__()
+        self.title = TitleText(title)
+
+    @property
+    def text(self):
+        return self.title.text
+
+    @text.setter
+    def text(self, value):
+        self.title.text = value
+
+    def compose(self):
+        yield CloseIcon()
+        yield self.title
+        yield TitleSpacer()
+
+
 class TogaWindow(TextualScreen):
-    def __init__(self, impl):
+    def __init__(self, impl, title):
         super().__init__()
         self.interface = impl.interface
         self.impl = impl
+        self.titlebar = TitleBar(title)
```

Comment:
I noticed that the standard title bar expands when you click on it, which isn't really useful. Was that the reason for replacing it, or was there something else?

Reply:
The primary reason for replacing it was that `Header()` can (AFAICT) only have the title of the app. That's great for the default title of a MainWindow, but useless for every other location where we might need a header (such as the titlebar of a dialog, or the titlebar or any other window - or changing the title of the main window. Removing the "thick" rendering was a bonus.